app/views/venues/show.html.erb

```diff
@@ -1,15 +1,27 @@
 <h1><%= @venue.venue_name %></h1>
 <p><%= @venue.description %></p>
+
+<% unless current_user.nil? %>
+  <% if current_user.admin? || @venue.user_id == current_user.id %>
+    <%= button_to "Delete Venue", venue_path, class: "button", method: "delete", label: "Delete Venue"%>
+  <% end %>
+<% end %>
 <ul>
   <% @reviews.each do |review| %>
-  <li> <%= review.review_body %></li>
-  <%= review.rating %>
+    <li> <%= review.review_body %></li>
+    <%= review.rating %>
+    <% unless current_user.nil? %>
+      <% if current_user.admin? || review.user_id == current_user.id %>
+        <%= button_to "Delete Review", "#{@venue.id}/reviews/#{review.id}", class: "button", method: "delete", label: "Delete Review" %>
+      <% end %>
+    <% end %>
   <% end %>
```

Comment:
Same comment here - notice the three different conditions you're checking, and notice that they're the same as for `delete` venue? You can implement the exact same method on `review.rb` as you did for `venue.rb` and avoid both compound conditions and nested control flow, cleaning up the view considerably